iquip/apps/builder.py

```diff
@@ -683,3 +709,118 @@ def onSubmitted(self, rid: int):
     def frames(self) -> Tuple[Tuple[str, BuilderFrame]]:
         """Overridden."""
         return (("", self.builderFrame),)
+
+
+class _RangeScan(QWidget):
+    """A RangeScan widget in Scannable Entry.
+
+    Attributes:
+        layout: The layout of _RangeScan widget.
+    """
+    def __init__(
+        self,
+        procdesc: Dict[str, Any],
+        state: Dict[str, Any],
+        parent: Optional[QWidget] = None
+        ): # pylint: disable=too-many-statements
+        """Extended.
+
+        Args:
+            procdesc: Each key and its value are as follows.
+              unit: The unit of the number value.
+              scale: The scale factor that is multiplied to the number value.
+              global_step: The step between values changed by the up and down button.
+              global_min: The minimum value. (default=float("-inf"))
+              global_max: The maximum value. (default=float("inf"))
+                If min > max, then they are swapped.
+              ndecimals: The number of displayed decimals.
+            state: Each key and its value as follows.
+              start: The start point for the RangeScan sequence.
+              stop: The end point for the RangeScan sequence.
+              npoints: The number of points in the RangeScan squence.
+              randomize: A boolean value that decides whether to shuffle the RangeScan sequence.
+        """
+        super().__init__(parent=parent)
+        scale = procdesc["scale"]
+
+        def apply_properties(widget):
```

Comment:
How about making it a method, not an internal function?

Also, please add the type hinting.